### PR TITLE
DM-47128: PyPI deployment and entry points

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -50,13 +50,77 @@ jobs:
...
           pytest -r a -v -n 3 --cov=lsst.dax.obscore --cov=tests --cov-report=xml --cov-report=term --cov-branch
+          butler obscore -h
 
       - name: Upload coverage to codecov
         uses: codecov/codecov-action@v4
         with:
           files: ./coverage.xml
           token: ${{ secrets.CODECOV_TOKEN }}
+
+  check-changes:
```
Comment (timj) [inline .github/workflows/build.yaml:63]: @mwittgen is the plan to put this code in a reusable workflow? It looks like we only made this change in sphgeom and not all the other middleware packages.

Comment (mwittgen) [reply .github/workflows/build.yaml:63]: Yes.

**Comment by mwittgen** [comment]
The workflow should run on tag
```
on:
  push:
    branches:
      - main
    tags:
      - "*"
```

**Comment by timj** [comment]
Thanks for spotting the missing tags section.

**Review by mwittgen — APPROVED** [review]
Looks good now. Quickly tested the upload check in my own fork.

**Pull request merged**